query-model/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/querymodel/jpql/expression/impl/JpqlSize.kt

```diff
@@ -0,0 +1,10 @@
+package com.linecorp.kotlinjdsl.querymodel.jpql.expression.impl
+
+import com.linecorp.kotlinjdsl.Internal
+import com.linecorp.kotlinjdsl.querymodel.jpql.expression.Expression
+import com.linecorp.kotlinjdsl.querymodel.jpql.path.Path
+
+@Internal
+data class JpqlSize internal constructor(
+    val path: Path<*>,
```

Comment:
I can not sure that this class should take Path or Expression

```
SIZE(collection_valued_path_expression)
```

Reply:
I think Path is a better fit than Expression, and I think we should limit the generic type of Path to Collection.

When I checked the Criteria API, it takes an Expression, but only takes into account the Path.
![image](https://github.com/line/kotlin-jdsl/assets/31586979/c42f77f8-63f8-4ff0-b4e1-b3d8d30db77e)